lib/core/src/Cardano/Wallet.hs

```diff
@@ -1192,8 +1195,23 @@ estimateFeeForPayment
 estimateFeeForPayment ctx wid recipients = do
     (utxo, txp) <- withExceptT ErrSelectForPaymentNoSuchWallet $
         selectCoinsSetup @ctx @s @k ctx wid
-    estimateFeeForCoinSelection $
+    let feePolicy@(LinearFee (Quantity a) _ _) = txp ^. #getFeePolicy
+    let feeOptions = (feeOpts tl (WithDelegation False) feePolicy)
+            { dustThreshold = Coin $ ceiling a }
+    let totalBalance = W.balance' $ Map.toList $ W.getUTxO utxo
+    let outs = sum $ map (\(TxOut _ (Coin c)) -> c) $ NE.toList recipients
+    let wholeSel =
+            CoinSelection
+            (Map.toList $ W.getUTxO utxo)
+            (NE.toList recipients)
+            [Coin (totalBalance - outs)]
+    let feeMin = getFee $ estimateFee feeOptions wholeSel
+    if (totalBalance - outs < feeMin && totalBalance >= outs) then
+            pure $ FeeEstimation feeMin feeMin
```

Comment:
I am still not convinced than this is the _right_ fix. We are patching on top of something that should actually not happened in the first place. Plus, I don't quite understand why this assertion links together the wallet's total balance and the transaction outputs? 

What this predicates seems to suggest a particular case that isn't properly handled in the coin selection algorithm when the change output is smaller than the minimal fee.

Reply:
right, I am also not 100% this is the best place - the place here https://github.com/input-output-hk/cardano-wallet/pull/1782#pullrequestreview-435072018 is better